src/main/java/org/ow2/proactive/workflow_catalog/rest/service/WorkflowRevisionService.java

```diff
@@ -277,13 +277,29 @@ private QueryExpressionContext createJpaQueryExpression(
         WorkflowMetadata workflowMetadata =
                 new WorkflowMetadata(workflowRevision);
 
-        workflowMetadata.add(createLink(bucketId, workflowId, revisionId, workflowRevision));
+        workflowMetadata.add(createLink(bucketId, workflowId, workflowRevision));
 
         return ResponseEntity.ok(workflowMetadata);
     }
 
-    public Link createLink(Long bucketId, Long workflowId, Optional<Long> revisionId,
-            WorkflowRevision workflowRevision) {
+    public ResponseEntity<?> delete(Long bucketId, Long workflowId, Optional<Long> revisionId) {
+        Workflow workflow = findWorkflow(workflowId);
```

Comment:
Very nice, I like that findWorkflow throws a runtime exception, avoiding null and try catch!